client-tez/src/main/java/org/apache/tez/common/UmbilicalUtils.java

```diff
@@ -0,0 +1,112 @@
+import org.apache.tez.dag.api.TezException;
+import org.apache.tez.dag.records.TezTaskAttemptID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.uniffle.common.ShuffleServerInfo;
+
+public class UmbilicalUtils {
+  private static final Logger LOG = LoggerFactory.getLogger(UmbilicalUtils.class);
+
+  private UmbilicalUtils() {
+  }
+
+  /**
+   *
+   * @return Get Application Master host and port from config file
+   */
+  private static Pair<String, Integer> getAmHostPort() {
+    JobConf conf = new JobConf(RssTezConfig.RSS_CONF_FILE);
+    String host = conf.get(RssTezConfig.RSS_AM_SHUFFLE_MANAGER_ADDRESS, "null host");
+    int port = conf.getInt(RssTezConfig.RSS_AM_SHUFFLE_MANAGER_PORT, -1);
+    LOG.info("Got RssConf am info, host is: {}, port is: {}", host, port);
+    return new ImmutablePair<>(host, port);
+  }
+
+  /**
+   *
+   * @param applicationId Application Id of this task
+   * @param conf  Configuration
+   * @param taskAttemptId task Attempt Id
+   * @param shuffleId   computed using dagId, up dagName, down dagName by RssTezUtils.computeShuffleId() method.
+   * @return Shuffle Server Info by request Application Master
+   * @throws IOException
+   * @throws InterruptedException
+   * @throws TezException
+   */
+  private static Map<Integer, List<ShuffleServerInfo>> doRequestShuffleServer(
+      ApplicationId applicationId,
+      Configuration conf,
+      TezTaskAttemptID taskAttemptId,
+      int shuffleId) throws IOException, InterruptedException, TezException {
+    UserGroupInformation taskOwner = UserGroupInformation.createRemoteUser(applicationId.toString());
+
+    Pair<String, Integer> amHostPort = getAmHostPort();
+    final InetSocketAddress address = NetUtils.createSocketAddrForHost(amHostPort.getLeft(), amHostPort.getRight());
+    TezRemoteShuffleUmbilicalProtocol umbilical = taskOwner
+        .doAs(new PrivilegedExceptionAction<TezRemoteShuffleUmbilicalProtocol>() {
+          @Override
+          public TezRemoteShuffleUmbilicalProtocol run() throws Exception {
+            return RPC.getProxy(TezRemoteShuffleUmbilicalProtocol.class,
+                TezRemoteShuffleUmbilicalProtocol.versionID,
+                address, conf);
+          }
+        });
+    GetShuffleServerRequest request = new GetShuffleServerRequest(taskAttemptId, 200, 200, shuffleId);
+
+    GetShuffleServerResponse response = umbilical.getShuffleAssignments(request);
+    Map<Integer, List<ShuffleServerInfo>>  partitionToServers = response.getShuffleAssignmentsInfoWritable()
+        .getShuffleAssignmentsInfo()
+        .getPartitionToServers();
+    LOG.info("RequestShuffleServer applicationId:{}, taskAttemptId:{}, host:{}, port:{}, shuffleId:{}, worker:{}",
+        applicationId, taskAttemptId, amHostPort.getLeft(), amHostPort.getRight(), shuffleId, partitionToServers);
+    return partitionToServers;
+  }
+
+  public static Map<Integer, List<ShuffleServerInfo>> requestShuffleServer(
```

Comment:
Could you add UT for this class?

Reply:
ok, it is not easy as early, we will add later.

Reply:
OK.